packages/api/src/controllers/webhook.ts

```diff
@@ -188,6 +196,48 @@ app.put("/:id", authorizer({}), validatePost("webhook"), async (req, res) => {
   res.json({ id: req.body.id });
 });
 
+app.patch(
+  "/:id",
+  authorizer({}),
+  validatePost("webhook-patch-payload"),
+  async (req, res) => {
+    const webhook = await db.webhook.get(req.params.id);
+
+    if (!webhook) {
+      throw new NotFoundError(`webhook not found`);
+    }
+
+    if (
+      (webhook.userId !== req.user.id || webhook.deleted) &&
+      !req.user.admin
+    ) {
+      // do not reveal that webhooks exists
+      throw new NotFoundError(`webhook not found`);
+    }
+
+    const { id, userId, createdAt, kind } = webhook;
+
+    if (req.body.streamId) {
+      const stream = await db.stream.get(req.body.streamId);
+      if (!stream) {
```

Comment:
Gotta check that the stream is not deleted and from the same user as well
```suggestion
      if (!stream || stream.deleted || stream.userId !== webhook.userId)  {
```

Reply:
Good catch!